src/graphql/resolvers/metrics.js

```diff
@@ -0,0 +1,167 @@
+const config = require('config');
+const { elasticClient } = require('../lib/elastic');
+const logger = require('../lib/logger');
+
+const metricsType = require('../models/metrics');
+
+const metrics = {
+  type: metricsType,
+  resolve: async (parent, args, req) => {
+    let index = req?.get('index');
+
+    if (!index) {
+      index = config.get('elasticsearch.indexAlias');
+    }
+
+    let res;
+
+    try {
+      res = await elasticClient.count({
+        index,
+      });
+    } catch (err) {
+      logger.error('Cannot request elastic');
+      logger.error(err);
+      return null;
+    }
+
+    const doi = res.body.count;
+
+    try {
+      res = await elasticClient.count({
+        index,
+        body: {
+          query: {
+            bool: {
+              filter: {
+                term: { is_oa: true },
+              },
+            },
+          },
+        },
+      });
+    } catch (err) {
+      logger.error('Cannot request elastic');
+      logger.error(err);
+      return null;
+    }
+
+    const isOA = res.body.count;
+
+    try {
+      res = await elasticClient.count({
+        index,
+        body: {
+          query: {
+            bool: {
+              filter: {
+                term: { oa_status: 'gold' },
+              },
+            },
+          },
+        },
+      });
```

Comment:
If your goal is to count all the different values of `oa_status`, you'd be better doing an `aggregation` instead of repeating multiple `count` requests.

Reply:
... x)